build.cake

```diff
@@ -116,20 +118,37 @@ Task("Publish")
     .WithCriteria(BuildSystem.IsRunningOnTeamCity)
     .Does(() =>
 {
-    NuGetPush($"{artifactsDir}Octostache.{nugetVersion}.nupkg", new NuGetPushSettings {
-        Source = "https://f.feedz.io/octopus-deploy/dependencies/nuget",
-        ApiKey = EnvironmentVariable("FeedzIoApiKey"),
-        SkipDuplicate = true,
+    var currentBranch = GitBranchCurrent(DirectoryPath.FromString(".")).FriendlyName;
+    var octopusServer = EnvironmentVariable("OctopusServerUrl");
+    var octopusApiKey = EnvironmentVariable("OctopusServerApiKey");
+    var space = EnvironmentVariable("OctopusServerSpaceName");
+    var octopusProjectName = "Octostache";
+
+    var nugetPackage = GetFiles($"{artifactsDir}Octostache.{nugetVersion}.nupkg");
+
+    // Current config for this repo doesn't generate prerelease tags, even if we're on a development/feature branch.
+    // Using the --overwrite-mode=IgnoreIfExists flag instructs the target Octopus server to ignore any attempted uploads with the same verison number.
+    // This decision is made server-side, so you'll still see log messages indicating the package was uploaded. Never fear, the push is discarded if the version already exists. 
+    // You can verify this by looking at the SHA1 and Published date of the package in the package feed: it won't change on subsequent pushes.
+    OctoPush(octopusServer, octopusApiKey, nugetPackage, new OctopusPushSettings 
+    {
+        ArgumentCustomization = args => args.Append("--overwrite-mode=IgnoreIfExists"),
+        Space = space 
     });
 
-    if (gitVersionInfo.PreReleaseTagWithDash == "")
-    {
-        NuGetPush($"{artifactsDir}Octostache.{nugetVersion}.nupkg", new NuGetPushSettings {
-            Source = "https://www.nuget.org/api/v2/package",
-            ApiKey = EnvironmentVariable("NuGetApiKey"),
-            SkipDuplicate = true,
-        });
-    }
+    // Config-as-Code doesn't yet support Automatic Release Creation, so do it manually
+    OctoCreateRelease(octopusProjectName, new CreateReleaseSettings {
+        ArgumentCustomization = args => args.Append($"--gitRef={currentBranch}"),
+        Server = octopusServer,
+        ApiKey = octopusApiKey,
+        ReleaseNumber = nugetVersion,
+        Space = space,
+        Packages = new Dictionary<string, string>
+        {
+            { "Octostache", nugetVersion }
```

Comment:
my suggestion would be to not creating a release from the build definition. I would say Nuke should end at producing an artifact and let teamcity create the release

Reply:
> let teamcity create the release
What's the difference between `build.ps1` creating the release, and a separate step in TeamCity creating it? I would prefer to just let Octopus create it when a new package is pushed, but that isn't currently supported with Config-as-Code

Reply:
ah righto sorry missed that. Hmm we've haven't considered auto create release in F&M. We have by default created release either in build.ps1 or in teamcity. My take on ps1 creating the release is so that we don't combine concern of deployment with building. My personal preference is that Nuke/build script should not concern itself what happens after a package has been generated. It's really good to use for defining the clean -> restore -> compile -> pack or test. but beyond that I think we'd be giving it too much responsibility. 

As for auto creating releases, that makes sense to me. chances are we would want to move towards that. I don't have a strong point of view on that one 🙂